ts/Stock/RangeSelector/RangeSelector.ts

```diff
@@ -365,16 +363,24 @@ class RangeSelector {
         // Update the chart
         if (!baseAxis) {
             // Axis not yet instantiated. Temporarily set min and range
-            // options and remove them on chart load (#4317).
+            // options and axes once defined and remove them on
+            // chart load (#4317 & #20529).
             baseXAxisOptions = splat(chart.options.xAxis)[0];
-            rangeSetting = baseXAxisOptions.range;
-            baseXAxisOptions.range = range;
-            minSetting = baseXAxisOptions.min;
-            baseXAxisOptions.min = rangeMin;
+            const axisRangeUpdateEvent = addEvent(
```

Comment:
An alternative and simple fix would be to modify `chart.userOptions.xAxis` (instead of `chart.options.xAxis`).

However that raises a question: why the previous versions used to contain "userOptions" (the reference) in `chart.options.xAxis`? Just to be on the same page: I prefer the current solution (different `chart.options.xAxis` and `chart.userOptions.xAxis` objects). Should we expect more similar issues @TorsteinHonsi? 🤔

Reply:
In principle, `userOptions` should never be modified, they should be what it says, the user-defined options. All business code should use `.options` because that is the merged and processed options to be used. Ideally I wanted to remove `userOptions` altogether, but there are some very few cases where we need to check whether the user-defined option is explicitly different from the default.

> Should we expect more similar issues?

@pawelfus We might. I have tackled a few of those during refactoring in the past.